### ISSUE TITLE
Class inherits a class defined just below fails in js side but no error shown in ts side

### ISSUE THREAD

**Issue by TadasPa** [Microsoft/TypeScript#6526]
TS:
```ts
class A extends B { }
class B { }
// no TS errors here
```
JS:
```js
var A = (function (_super) {
        __extends(A, _super);
        function A() {
            _super.apply(this, arguments);
        }
        return A;
    })(B);
var B = (function () {
        function B() {
        }
        return B;
    })();
```
When "__extends(A, _super);" is called "_super" is not defined yet and it's prototype, of course. I guess TS side should complain or JS should be reordered.

**Comment by RyanCavanaugh** [comment]
There are about eleventybillion dupes on this but I thought we had fixed this already. Vlad says we can probably use `isBlockScopedNameDeclaredBeforeUse` to implement this very cheaply.

**Comment by drake7707** [comment]
Also see #4341

**Comment by mhegazy** [comment]
closing in favor of https://github.com/Microsoft/TypeScript/issues/5207
